sdk/inc/azure/core/az_http.h

```diff
@@ -26,100 +26,111 @@
+  AZ_HTTP_STATUS_CODE_PRECONDITION_FAILED = 412, ///< HTTP 412 Precondition Failed.
+  AZ_HTTP_STATUS_CODE_PAYLOAD_TOO_LARGE = 413, ///< HTTP 413 Payload Too Large.
+  AZ_HTTP_STATUS_CODE_URI_TOO_LONG = 414, ///< HTTP 414 URI Too Long.
+  AZ_HTTP_STATUS_CODE_UNSUPPORTED_MEDIA_TYPE = 415, ///< HTTP 415 Unsupported Media Type.
+  AZ_HTTP_STATUS_CODE_RANGE_NOT_SATISFIABLE = 416, ///< HTTP 416 Range Not Satisfiable.
+  AZ_HTTP_STATUS_CODE_EXPECTATION_FAILED = 417, ///< HTTP 417 Expectation Failed.
+  AZ_HTTP_STATUS_CODE_MISDIRECTED_REQUEST = 421, ///< HTTP 421 Misdirected Request.
+  AZ_HTTP_STATUS_CODE_UNPROCESSABLE_ENTITY = 422, ///< HTTP 422 Unprocessable Entity.
+  AZ_HTTP_STATUS_CODE_LOCKED = 423, ///< HTTP 423 Locked.
+  AZ_HTTP_STATUS_CODE_FAILED_DEPENDENCY = 424, ///< HTTP 424 Failed Dependency.
+  AZ_HTTP_STATUS_CODE_TOO_EARLY = 425, ///< HTTP 425 Too Early.
+  AZ_HTTP_STATUS_CODE_UPGRADE_REQUIRED = 426, ///< HTTP 426 Upgrade Required.
+  AZ_HTTP_STATUS_CODE_PRECONDITION_REQUIRED = 428, ///< HTTP 428 Precondition Required.
+  AZ_HTTP_STATUS_CODE_TOO_MANY_REQUESTS = 429, ///< HTTP 429 Too Many Requests.
+  AZ_HTTP_STATUS_CODE_REQUEST_HEADER_FIELDS_TOO_LARGE
+  = 431, ///< HTTP 431 Request Header Fields Too Large.
+  AZ_HTTP_STATUS_CODE_UNAVAILABLE_FOR_LEGAL_REASONS
+  = 451, ///< HTTP 451 Unavailable For Legal Reasons.
 
   // 5xx (server error) Status Codes:
-  AZ_HTTP_STATUS_CODE_INTERNAL_SERVER_ERROR = 500,
-  AZ_HTTP_STATUS_CODE_NOT_IMPLEMENTED = 501,
-  AZ_HTTP_STATUS_CODE_BAD_GATEWAY = 502,
-  AZ_HTTP_STATUS_CODE_SERVICE_UNAVAILABLE = 503,
-  AZ_HTTP_STATUS_CODE_GATEWAY_TIMEOUT = 504,
-  AZ_HTTP_STATUS_CODE_HTTP_VERSION_NOT_SUPPORTED = 505,
-  AZ_HTTP_STATUS_CODE_VARIANT_ALSO_NEGOTIATES = 506,
-  AZ_HTTP_STATUS_CODE_INSUFFICIENT_STORAGE = 507,
-  AZ_HTTP_STATUS_CODE_LOOP_DETECTED = 508,
-  AZ_HTTP_STATUS_CODE_NOT_EXTENDED = 510,
-  AZ_HTTP_STATUS_CODE_NETWORK_AUTHENTICATION_REQUIRED = 511,
-
-  // Used in az_http_policy_retry_options to indicate the end of the list
-  AZ_HTTP_STATUS_CODE_END_OF_LIST = -1,
+  AZ_HTTP_STATUS_CODE_INTERNAL_SERVER_ERROR = 500, ///< HTTP 500 Internal Server Error.
+  AZ_HTTP_STATUS_CODE_NOT_IMPLEMENTED = 501, ///< HTTP 501 Not Implemented.
+  AZ_HTTP_STATUS_CODE_BAD_GATEWAY = 502, ///< HTTP 502 Bad Gateway.
+  AZ_HTTP_STATUS_CODE_SERVICE_UNAVAILABLE = 503, ///< HTTP 503 Unavailable.
+  AZ_HTTP_STATUS_CODE_GATEWAY_TIMEOUT = 504, ///< HTTP 504 Gateway Timeout.
+  AZ_HTTP_STATUS_CODE_HTTP_VERSION_NOT_SUPPORTED = 505, ///< HTTP 505 HTTP Version Not Supported.
+  AZ_HTTP_STATUS_CODE_VARIANT_ALSO_NEGOTIATES = 506, ///< HTTP 506 Variant Also Negotiates.
+  AZ_HTTP_STATUS_CODE_INSUFFICIENT_STORAGE = 507, ///< HTTP 507 Insufficient Storage.
+  AZ_HTTP_STATUS_CODE_LOOP_DETECTED = 508, ///< HTTP 508 Loop Detected.
+  AZ_HTTP_STATUS_CODE_NOT_EXTENDED = 510, ///< HTTP 510 Not Extended.
+  AZ_HTTP_STATUS_CODE_NETWORK_AUTHENTICATION_REQUIRED
+  = 511, ///< HTTP 511 Network Authentication Required.
```

Comment:
Now they are documented, previously they were not. I find HTTP specific number to be specifically useful, because, maybe it's just me, but I feel more confident when I write the HTTP code and know it matches the integral number that I am expecting.

What do you mean by capitalization?

Reply:
> What do you mean by capitalization?

For example, maybe change this to be lower-cased:
> HTTP 511 Network Authentication Required.

HTTP 511 Network authentication required.